src/Language/GraphQL/Draft/Parser.hs

```diff
@@ -479,3 +488,41 @@ between open close p = tok open *> p <* tok close
 -- `empty` /= `pure mempty` for `Parser`.
 optempty :: Monoid a => Parser a -> Parser a
 optempty = option mempty
+
+-- | Parses strings delimited by triple quotes. 
+-- http://spec.graphql.org/June2018/#sec-String-Value
+blockString :: Parser Text
+blockString = do
+  _ <- tripleQuotes <?> "opening triple quotes"
+  lines_ <- T.lines . T.pack <$> AT.manyTill AT.anyChar tripleQuotes <?> "the body of a triple quoted string"
+  let headline = if lines_ == [] then "" else head lines_
+  let tail_ = drop 1 lines_ -- not tail
```

Comment:
For some reason these two lines seem quite sketchy to me. As a start, we could do them simultaneously with a pattern match. But, more worryingly, this might be introducing some extra edge cases.
What's the role of `headline`? I suppose it's got something to do with the special treatment of the first line by `BlockStringValue` - if so, let's make this explicit, at least with a comment.
`tail_` is not a very helpful variable name. Is this the possibly-indented remainder of the block string? Then we could call it `indentedRemainder`.

Reply:
> Is this the possibly-indented remainder of the block string?

I think it is the non-indented remainder of the block string on which the common indentation is to be removed, point 4 [here](http://spec.graphql.org/June2018/#BlockStringValue())